internal/tlsutil/tls.go

```diff
@@ -20,12 +20,47 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"io/ioutil"
+	"strings"
 
 	"github.com/pkg/errors"
 )
 
-// NewClientTLS returns tls.Config appropriate for client auth.
+type TLSRenegotiationStrategy string
```

Comment:
Why are we defining a type, then never using it? Can we make `GetRenegotiation` accept a type `TLSRenegotiationStrategy`, and use this type for the renegotiation strategies `RenegotiateNever`, `RenegotiateOnceAsClient` and `RenegotiateFreelyAsClient`?

Here's a good reference on applying enum types: https://www.ribice.ba/golang-enums/

Reply:
I think that was my original thought but that would mean GetRenegotiation wouldn't accept a case insensitive string for the strategy. Should I implement that and move the  ToLower and type cast to somewhere else?

Reply:
yeah. The calls to `ToLower` and the type cast can be moved over to `cmd`, after parsing the flags.

In fact, you could introduce a function that parses a string and returns a `TLSRenegotiationStrategy`. Many logging packages use this to parse strings like "debug" or "info" for log levels.

logrus is a good example that demonstrates this pattern. Have a look at https://github.com/sirupsen/logrus/blob/53000c4c0f981521df1bada12a9e9ea20728fb62/logrus.go#L12-L112